client/jetpack-cloud/sections/manage/pricing/utils/get-api-family-product-icon.tsx

```diff
@@ -0,0 +1,93 @@
+import JetpackProductIconScan from 'calypso/assets/images/jetpack/jetpack-product-icon-scan.svg';
+import JetpackProductIconSearchLight from 'calypso/assets/images/jetpack/jetpack-product-icon-search-light.svg';
+import JetpackProductIconSearch from 'calypso/assets/images/jetpack/jetpack-product-icon-search.svg';
+import JetpackProductIconSocialLight from 'calypso/assets/images/jetpack/jetpack-product-icon-social-light.svg';
+import JetpackProductIconSocial from 'calypso/assets/images/jetpack/jetpack-product-icon-social.svg';
+import JetpackProductIconStatsLight from 'calypso/assets/images/jetpack/jetpack-product-icon-stats-light.svg';
+import JetpackProductIconStats from 'calypso/assets/images/jetpack/jetpack-product-icon-stats.svg';
+import JetpackProductIconVideopressLight from 'calypso/assets/images/jetpack/jetpack-product-icon-videopress-light.svg';
+import JetpackProductIconVideopress from 'calypso/assets/images/jetpack/jetpack-product-icon-videopress.svg';
+
+type productIconProps = {
+	productSlug: string;
+	light?: boolean;
+};
+
+const setProductsIcon = ( slug: string, resource: IconResource ) => ( {
+	[ slug ]: resource,
+} );
+
+interface IconResource {
+	light: string;
+	regular: string;
+}
+
+const PRODUCT_ICON_MAP: Record< string, IconResource > = {
+	...setProductsIcon( 'jetpack-backup', {
+		regular: JetpackProductIconBackup,
+		light: JetpackProductIconBackupLight,
+	} ),
+	...setProductsIcon( 'jetpack-ai', {
+		regular: JetpackProductIconAI,
+		light: JetpackProductIconAILight,
+	} ),
+	...setProductsIcon( 'jetpack-anti-spam', {
+		regular: JetpackProductIconAntiSpam,
+		light: JetpackProductIconAntiSpamLight,
+	} ),
+	...setProductsIcon( 'jetpack-scan', {
+		regular: JetpackProductIconScan,
+		light: JetpackProductIconScanLight,
+	} ),
+	...setProductsIcon( 'jetpack-videopress', {
+		regular: JetpackProductIconVideopress,
+		light: JetpackProductIconVideopressLight,
+	} ),
+	...setProductsIcon( 'jetpack-search', {
+		regular: JetpackProductIconSearch,
+		light: JetpackProductIconSearchLight,
+	} ),
+	...setProductsIcon( 'jetpack-boost', {
+		regular: JetpackProductIconBoost,
+		light: JetpackProductIconBoostLight,
+	} ),
+	...setProductsIcon( 'jetpack-creator', {
+		regular: JetpackProductIconCreator,
+		light: '',
+	} ),
+	...setProductsIcon( 'jetpack-monitor', {
+		regular: JetpackProductIconMonitor,
+		light: JetpackProductIconMonitorLight,
+	} ),
+	...setProductsIcon( 'jetpack-social-basic', {
+		regular: JetpackProductIconSocial,
+		light: JetpackProductIconSocialLight,
+	} ),
+	...setProductsIcon( 'jetpack-social-advanced', {
+		regular: JetpackProductIconSocial,
+		light: JetpackProductIconSocialLight,
+	} ),
+	...setProductsIcon( 'jetpack-stats', {
+		regular: JetpackProductIconStats,
+		light: JetpackProductIconStatsLight,
+	} ),
+};
+
+const getAPIFamilyProductIcon = ( { productSlug, light }: productIconProps ): string => {
```

Comment:
This looks quite similar to `client/my-sites/plans/jetpack-plans/product-store/utils/get-product-icon.ts`...in an ideal world, we'd avoid repeating ourselves.

Reply:
Originally I attempted to change this file directly, but found I was virtually doubling everything up to create a new PRODUCT_ICON_MAP const for this use case, and change getAPIFamilyProductIcon to accept a string as well as productIconProps, then check for either.  Possibly marginally less duplication by doing it this way - certainly less duplication on the imports and less code generally having two files rather than one 🤔